### DISCUSSION TITLE
Example for user & password for MQTT

### DISCUSSION THREAD

**Discussion by pmartinez-tst** [emqx/emqx#8127] [Q&A]
I have tried to add user and password for the MQTT clients but I do not fully understand ACL rules. I just want to have 1 user and password for all the clients or just 2. Can you show me an example?

Right now I see that I can access to my server without user and password and I want to avoid that.

I rembember that there was a file in old versions called emqx_auth_usename.conf or something like that, isn't it?

**Comment by techmantel** [comment]
>Right now I see that I can access to my server without user and password and I want to avoid that.

This is done by settings the `allow_anonymous=false` in `etc/emqx.conf` or through environment variable like so: `EMQX_ALLOW_ANONYMOUS=false`

I'm new to this software and I'd also like to see a basic example on how to do this. What I finally got to work was to use the REST API to add both user/app and ACLs to make it work.

So basically what I did was:

Enable the `emqx_auth_mnesia` plugin using the Dashboard.
Then send these commands to the HTTP API using e.g. postman.
```
POST /api/v4/auth_username
[
    {
        "username": "my_user",
        "password": "my_passwd"
    }
]
```
and then
```
POST /api/v4/acl
{
  "topic":"/my/topic",
  "action":"pub",
  "access": "allow"
}
```
after this my app could authenticate using the given credentials and publish to the topic.
However, I wonder why this is not supported in the dashboard, or is it? Nothing related to user management seems to affect these things there, but from what I can read from the documentation ([here](https://www.emqx.io/docs/en/v4.3/getting-started/dashboard.html#applications) and [here](https://www.emqx.io/docs/en/v4.3/getting-started/dashboard.html#users)) it's more for accessing the dashboard and HTTP API, right?

**Reply by zmstone** [reply]
Hi @techmantel
We just release version 5.0.0.
username and ACL are both managed from dashboard UI (and REST API too).
